framework/docker/cosmos/chain.go

```diff
@@ -449,35 +454,82 @@ func (c *Chain) pullImages(ctx context.Context) {
 	}
 }
 
-// CreateWallet creates a new wallet using Validator[0] to maintain backward compatibility.
+// CreateWallet creates a new wallet on the first node and copies the key to all other nodes.
 func (c *Chain) CreateWallet(ctx context.Context, keyName string) (*types.Wallet, error) {
-	return c.GetNode().CreateWallet(ctx, keyName, c.Config.Bech32Prefix)
+	wallet, err := c.GetNode().CreateWallet(ctx, keyName, c.Config.Bech32Prefix)
+	if err != nil {
+		return nil, err
+	}
+
+	// the faucet wallet is handled separately during chain initialization.
+	// when we are creating the faucet wallet, the other nodes will not be started yet
+	// and so cannot copy the key to them.
+	if keyName != consts.FaucetAccountKeyName {
+		if err := c.copyWalletKeyToAllNodes(wallet); err != nil {
+			return nil, fmt.Errorf("failed to copy wallet key to all nodes: %w", err)
+		}
+	}
+
+	return wallet, nil
 }
 
-// copyFaucetKeyToValidator copies the faucet key from validator[0] to the specified validator.
-func (c *Chain) copyFaucetKeyToValidator(faucetWallet *types.Wallet, targetValidator *ChainNode) error {
-	faucetKeyName := faucetWallet.GetKeyName()
+// copyWalletKeyToAllNodes copies a wallet's key from the first node to all other nodes in the chain.
+func (c *Chain) copyWalletKeyToAllNodes(wallet *types.Wallet) error {
+	nodes := c.Nodes()
+	if len(nodes) <= 1 {
+		return nil
+	}
+
+	for i := 1; i < len(nodes); i++ {
+		if err := c.ensureNodeKeyringInitialized(nodes[i]); err != nil {
+			return fmt.Errorf("failed to initialize keyring for node %d: %w", i, err)
+		}
+
+		if err := c.copyWalletToValidator(wallet, nodes[i]); err != nil {
+			return fmt.Errorf("failed to copy wallet key to node %d: %w", i, err)
+		}
+	}
+
+	return nil
+}
+
+// ensureNodeKeyringInitialized ensures the keyring directory exists on the target node by creating a temporary key if needed.
+func (c *Chain) ensureNodeKeyringInitialized(node *ChainNode) error {
+	kr, err := node.GetKeyring()
+	if err == nil {
+		if _, err := kr.List(); err == nil {
+			return nil
+		}
+	}
+
+	dummyKeyName := "temp-init-key"
```

Comment:
Nit: Is this dummyKeyName stored as the key name on the node? if so, we probably should make sure that we won't reuse this name in future again. Not sure what is the best way of doing it.

Reply:
yeah, I don't really like this solution tbh, let me see if I can just make the file structure directly. If not we can append a random element to the keyname

Reply:
ah confirmed just making the directory is enough